devlib/trace/perf.py

```diff
@@ -113,29 +143,67 @@ def stop(self):
     # pylint: disable=arguments-differ
     def get_trace(self, outdir):
         for label in self.labels:
-            target_file = self._get_target_outfile(label)
-            host_relpath = os.path.basename(target_file)
-            host_file = _f(os.path.join(outdir, host_relpath))
-            self.target.pull(target_file, host_file)
+            # Sometime perf/simpleperf takes a while to write to .data file after kill signal
+            time.sleep(5)
+            if self.command == 'record':
+                report_command = self._build_perf_report_command(self.report_options, label)
+                self.target.kick_off(report_command, as_root=True)
+                # For larger records, report file takes a while to generate
+                time.sleep(5)
```

Comment:
Remove long fixed-length sleep -- wait on a condition (e.g. probe in a loop with a short sleep between iterations).